torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1863,6 +1876,35 @@ def _lazy_init(self) -> None:
         for handle in self._handles:
             self._init_param_attributes(handle)
         self._exec_order_data.init(self, self.process_group)
+        # Initialize inter-node process group for hybrid shard comm. We assume
```

Comment:
Just as my 2 cents that might help transition, to prepare for rebasing and/or after rebasing, we should
- refactor this inter-node process group initialization into its own helper method like `_init_inter_node_process_groups(state: _FSDPState)` that goes in `_init_utils.py`
- only call `_init_inter_node_process_groups()` in `FullyShardedDataParallel.__init__()` (i.e. no need for composable FSDP)